src/extension.js

```diff
@@ -0,0 +1,263 @@
+// noinspection JSUnusedGlobalSymbols
+const extension = {
+  options: {
+    sync: new OptionSection(
+      chrome.storage.sync,
+      {
+        wrapNavigation: false,
+        autoSelectFirst: true,
+        nextKey: 'down, j',
+        previousKey: 'up, k',
+        navigatePreviousResultPage: 'left, h',
+        navigateNextResultPage: 'right, l',
+        navigateKey: 'return, space',
+        navigateNewTabKey: 'ctrl+return, command+return, ctrl+space',
+        navigateSearchTab: 'a, s',
+        navigateImagesTab: 'i',
+        navigateVideosTab: 'v',
+        navigateMapsTab: 'm',
+        navigateNewsTab: 'n',
+        focusSearchInput: '/, escape'
+      }
+    ),
+
+    local: new OptionSection(
+      chrome.storage.local,
+      {
+        lastQueryUrl: null,
+        lastFocusedIndex: 0
+      }
+    ),
+
+    load: function() {
+      return Promise.all([this.local.load(), this.sync.load()]);
+    },
+  },
+
+  init: function() {
+    if (!/^(www|encrypted)\.google\./.test(window.location.hostname)) {
+      return;
+    }
+
```

Comment:
Here and below: can you please decrease extra newlines/whitespace inside functions?